prime-angular-frontend/src/app/modules/adjudication/pages/health-authorities/vendors-page/vendors-page.component.ts

```diff
@@ -92,21 +113,23 @@ export class VendorsPageComponent implements OnInit {
 
   private initForm() {
     this.form.valueChanges
-      .subscribe(({ vendors }: { vendors: { vendor: VendorConfig }[] }) => {
-        const selectedVendorCodes = vendors.map(ct => ct.vendor?.code);
+      .subscribe(() => {
+        const selectedVendorCodes = this.vendors.getRawValue().map(v => v.vendor?.code);
         // Filter out the selected vendors to avoid visual duplicates
         const filteredVendors = this.healthAuthorityVendors.filter(v => !selectedVendorCodes.includes(v.code));
         this.filteredVendors.next(filteredVendors);
       });
 
     this.healthAuthResource.getHealthAuthorityById(this.route.snapshot.params.haid)
-      .subscribe(({ vendors }: HealthAuthority) =>
-        (vendors?.length)
-          ? this.configService.vendors
-            .filter(v => vendors.some(({ vendorCode }) => v.code === vendorCode))
-            .map(v => this.addVendor(v))
-          : this.addVendor()
-      );
+      .subscribe(({ vendors }: HealthAuthority) => {
+        vendors
```

Comment:
So like you have in care setting you will need the check on length, since this currently doesn't work when you are filling out a new HA

Reply:
Ah ok, that was the case where I was wondering if we'd ever have no vendors.